From: Sven Eckelmann <sven.eckelmann@open-mesh.com>
Date: Fri, 8 Jul 2016 11:30:27 +0200
Subject: batman-adv: Exchange include order for uapi header files

The build of netlink.c in batman-adv 2016.3 will fail because the
definition of GENL_NAMESIZE and similar things are missing. This is is
caused by an incorrect order of backports and non-backports header files
mixed up with uapi and non-uapi header files. Parts of the header files
would not get included by include_next because it not anymore in the
remaining search paths.

Reported: Marek Lindner <marek.lindner@open-mesh.com>
Signed-off-by: Sven Eckelmann <sven.eckelmann@open-mesh.com>

Origin: upstream, https://github.com/openwrt-routing/packages/commit/71f835fdeb2a7c37cdc3a5228e7ebc0e76efc8bd.patch

diff --git a/batman-adv/Makefile b/batman-adv/Makefile
index c45cbb4..1668083 100644
--- a/batman-adv/Makefile
+++ b/batman-adv/Makefile
@@ -62,10 +62,10 @@ PKG_EXTRA_CFLAGS:= \
 
 NOSTDINC_FLAGS = \
 	-I$(PKG_BUILD_DIR)/net/batman-adv \
-	-I$(STAGING_DIR)/usr/include/mac80211-backport/uapi \
 	-I$(STAGING_DIR)/usr/include/mac80211-backport \
-	-I$(STAGING_DIR)/usr/include/mac80211/uapi \
+	-I$(STAGING_DIR)/usr/include/mac80211-backport/uapi \
 	-I$(STAGING_DIR)/usr/include/mac80211 \
+	-I$(STAGING_DIR)/usr/include/mac80211/uapi \
 	-include backport/backport.h \
 	-include $(PKG_BUILD_DIR)/compat-hacks.h
 
